src/components/shared/TokenSelector/TokenSelector.tsx

```diff
@@ -61,51 +55,58 @@ const getStatusText = ({
     };
   }
 
-  if (hasError || noTokenData) {
+  if (hasError || !token) {
     return {};
   }
 
+  const { name, symbol } = token;
   return {
     status: MSG.preview,
     statusValues: { name, symbol },
   };
 };
 
-const getTokenData = (tokenQuery?: GetTokenByAddressQuery) => {
-  const tokenData = tokenQuery?.getTokenByAddress?.items;
-  const { name, symbol } = tokenData?.[0] || {};
-  return { name, symbol };
-};
-
 const getLoadingState = (isLoading: boolean) =>
   // Don't show loading state if connection is 4G or unknown
```

Comment:
@willm30 I can't remember what was the reason behind this, do you remember why we're not showing loading state for 4g connections?

Reply:
Showing the loading state for a query that returns quickly looks bad IMO, since it flashes momentarily then disappears. I'm not tied to this way of doing it but I do think it's an improvement over the flashing